### PR TITLE
[DEVCON-6862] Execute terraform workflow from adhoc server mode

### PR DESCRIPTION
**Pull request by smonero**
Just calls ExecuteWorkflow(). Plumbs the values needed from the globalCfg to the func `ConstructAdhocExecParams`. Also realized we don't actually need them in the config or server so I removed those instances.

relies on the other 2 PRs where we change things to use the pr number instead of the revision

### CODE REVIEW

File: server/neptune/adhoc/server.go [hunk 1]
```diff
@@ -191,6 +226,12 @@ func (s Server) Start() error {
 		s.Logger.InfoContext(ctx, "Shutting down terraform worker, resource clean up may still be occurring in the background")
 	}()
 
+	_, err := s.manuallyExecuteTerraformWorkflow(s.adhocExecutionParams)
```
Comment (smonero) [inline server/neptune/adhoc/server.go:229]: Do we care about the result here?

**Review by samrabelachew — COMMENTED** [review]
Let's break this up into separate PRs, one enabling the workflow, the other changing the mode name

**Comment by smonero** [comment]
Q's and answers:
- Do we care about the result from manually executing the workflow? - no, that would only be if we wanted to interact with it during or after execution
- Should it be in its own go routine? - yes
- Should we reuse the root_config_builder from the gateway config? (We have our own config, but does that mean we should also have our own root_config_builder (we need this to get the atlantis root from) (we talked about making our own since we wanted to potentially improve the "double-clone" issue but not sure if that was concrete) - yes, we need our own, this can be a separate PR
- Was thinking about breaking down this PR into 3 PRs: 1 to make a function that gets all the ghRepo values we need, 1 to make a function that makes the atlantisRoot we need, then a 3rd one that takes those and constructs the request that it then passes to the ExecuteWorkflow func. Thoughts? - ok

**Comment by smonero** [comment]
will rely on https://github.com/lyft/atlantis/pull/735

**Comment by smonero** [comment]
now failing with `[4/12/2024, 2:22:39 PM] Atlantis started - listening on port 4141
[4/12/2024, 2:22:39 PM] Started Worker
[4/12/2024, 2:22:39 PM] wrote git credentials to /root/.git-credentials
[4/12/2024, 2:22:39 PM] successfully ran git config --global credential.helper store
[4/12/2024, 2:22:39 PM] successfully ran git config --global url.https://x-access-token@github.com.insteadOf ssh://git@github.com
[4/12/2024, 2:22:39 PM] building root cfgs: creating temporary clone at path: : failed to clone directory: running command in separate process group: waiting for process: exit status 128`

**Comment by smonero** [comment]
ok fixed, now next issue is invalid file fetcher options

`[4/14/2024, 12:12:04 PM] Atlantis started - listening on port 4141
[4/14/2024, 12:12:04 PM] Started Worker
[4/14/2024, 12:12:05 PM] wrote git credentials to /root/.git-credentials
[4/14/2024, 12:12:05 PM] successfully ran git config --global credential.helper store
[4/14/2024, 12:12:05 PM] successfully ran git config --global url.https://x-access-token@github.com.insteadOf ssh://git@github.com
[4/14/2024, 12:12:05 PM] building root cfgs: getting matching roots: finding modified files: []: invalid fileFetcherOptions
`

**Comment by smonero** [comment]
will remove all the logging tomorrow once we get the other prs merged and this tested

**Comment by smonero** [comment]
cool it works!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Review by samrabelachew — APPROVED** [review]

**Pull request merged**